compiler/rustc_codegen_ssa/src/mir/place.rs

```diff
@@ -112,8 +112,7 @@ impl<'a, 'tcx, V: CodegenObject> PlaceRef<'tcx, V> {
                     if offset == a.size(bx.cx()).align_to(b.align(bx.cx()).abi) =>
                 {
                     // Offset matches second field.
-                    let ty = bx.backend_type(self.layout);
-                    bx.struct_gep(ty, self.llval, 1)
+                    bx.inbounds_gep(bx.type_i8(), self.llval, &[bx.const_usize(offset.bytes())])
```

Comment:
Agreed, I've dropped the check. This slightly changes the behavior because we now generate inbounds for the ScalarPair ZST case, which was previously using the case below. I could reorder the cases to avoid this, but using inbounds here is correct.

I've also left a FIXME to always generate this GEP regardless of ABI (#98615). I think we at this point we're ready to make that change.